plugins/mission/mission_impl.cpp

```diff
@@ -932,4 +934,135 @@ void MissionImpl::process_timeout()
     }
 }
 
+Mission::Result
+MissionImpl::import_qgroundcontrol_mission(Mission::mission_items_t &mission_items,
+                                           const std::string &qgc_plan_file)
+{
+    std::ifstream file(qgc_plan_file);
+    if (!file) { // File open error
+        return Mission::Result::FAILED_TO_OPEN_QGC_PLAN;
+    }
+
+    // Read QGC plan into a string stream
+    std::stringstream ss;
+    ss << file.rdbuf();
+    file.close();
+
+    std::string err;
+    const auto parsed_plan = Json::parse(ss.str(), err); // parse QGC plan
+    if (!err.empty()) { // Parse error
+        return Mission::Result::FAILED_TO_PARSE_QGC_PLAN;
+    }
+
+    // Compose mission items
+    return compose_mission_items(mission_items, parsed_plan);
+}
+
+Mission::Result MissionImpl::compose_mission_items(MAV_CMD cmd, std::vector<double> params,
```

Comment:
The name of this method is confusing. When I read the name I don't understand what it does and why it requires these arguments.

Reply:
First of all, its a static method; hence it doesn't deal with **this** and hence it doesn't modify the state of the mission object.
It is intended to compose mission items into an OUT parameter `mission_items` (last one). It builds a mission item out of the first two parameters it receives. 
So every invocation will build one mission item and add it to mission_items vector. 
Probably I can name this as `compose_each_mission_item()` ?

Reply:
I think my choice with `compose` was bad because it doesn't reflect what is going on.

> It builds a mission item out of the first two parameters it receives.
> So every invocation will either build one mission item or adds actions to the last mission item and when finally done

So could you split it up into two methods named: `build_a_mission_item(cmd, params)` and `add_to_mission_items(mission_item, all_mission_items)`?

Something like that would be clearer.

Reply:
I wanted to so. But, `build_a_mission_item()` also involves [adding to mission item list](https://github.com/dronecore/DroneCore/pull/235/files#diff-3a666f956b3fd53eaf1268ec05ea8d29R973). Its like `assemble_mavlink_mission_items()`. In fact, I did refactoring for my case.